wandb/apis/public.py

```diff
@@ -662,8 +736,268 @@ def __next__(self):
+        """
+    mutation CreateInvite($entityName: String!, $email: String, $username: String, $admin: Boolean) {
+        createInvite(input: {entityName: $entityName, email: $email, username: $username, admin: $admin}) {
+            invite {
+                id
+                name
+                email
+                createdAt
+                toUser {
+                    name
+                }
+            }
+        }
+    }
+    """
+    )
+    TEAM_QUERY = gql(
+        """
+    query Entity($name: String!) {
+        entity(name: $name) {
+            id
+            name
+            available
+            photoUrl
+            readOnly
+            readOnlyAdmin
+            isTeam
+            privateOnly
+            storageBytes
+            codeSavingEnabled
+            defaultAccess
+            isPaid
+            members {
+                id
+                admin
+                pending
+                email
+                username
+                name
+                photoUrl
+                accountType
+                apiKey
+            }
+        }
+    }
+    """
+    )
+    CREATE_SERVICE_ACCOUNT_MUTATION = gql(
+        """
+    mutation CreateServiceAccount($entityName: String!, $description: String!) {
+        createServiceAccount(
+            input: {description: $description, entityName: $entityName}
+        ) {
+            user {
+                id
+            }
+        }
+    }
+    """
+    )
+
+    def __init__(self, client, name, attrs=None):
+        super(Team, self).__init__(attrs or {})
+        self._client = client
+        self.name = name
+        self.load()
+
+    @classmethod
+    def create(cls, api, team, admin_username=None):
+        """Creates a new team
+
+        Arguments:
+            api: (`Api`) The api instance to use
+            team: (str) The name of the team
+            admin_username: (str) optional username of the admin user of the team, defaults to the current user.
+
+        Returns:
+            A `Team` object
+
+        Raises:
+            requests.exceptions.HTTPError if the team already exists
+        """
+        api.execute(
+            cls.CREATE_TEAM_MUTATION,
+            {"teamName": team, "teamAdminUserName": admin_username},
+        )
+        return Team(api, team)
+
+    def invite(self, username_or_email, admin=False):
+        """Invites a user to a team
+
+        Arguments:
+            username_or_email: (str) The username or email address of the user you want to invite
+            admin: (bool) Whether to make this user a team admin, defaults to False
+
+        Returns:
+            True on success
+
+        Raises:
+            requests.exceptions.HTTPError if the user couldn't be found
+        """
+        variables = {"entityName": self.name, "admin": admin}
+        if "@" in username_or_email:
+            variables["email"] = username_or_email
+        else:
+            variables["username"] = username_or_email
```

Comment:
corner case: if username has `@` do we check for this (i.e. is this allowed?)

Reply:
Usernames can only include letters, dashes, and underscores.

Reply:
got it thanks. 
just for my understanding: If someone uses our public api and provides invalid username - are we only catching it in the backend?

Reply:
We should be preventing it on user creation and we don't provide an api's to create users currently.